### PR TITLE
fix: webhook source integration tests

### PR DESCRIPTION
**Pull request by sanpj2292**
# Description

In current webhook source integration tests we don't have the following 
- capability of reading query parameter
- Put anonymousId received / generated into expected payload
- Insert writeKey into `properties.writeKey`
- We are also making sure the configuration destination types are in same casing
- Capability of running tests in local by hitting local transformer for source transformation

## Linear Ticket

Resolves INT-2353

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: gateway/webhook/integration_test.go [hunk 1]
```diff
@@ -105,12 +105,18 @@ func TestIntegrationWebhook(t *testing.T) {
 		application        app.App
 	)
 
+	transformerURL := "http://localhost:9090"
+	isLocal := config.GetBool("WEBHOOK_INTEGRATION_TEST_LOCAL", false)
+	if transformerContainer.TransformerURL != "" && !isLocal {
+		transformerURL = transformerContainer.TransformerURL
+	}
```
Comment (lvrach) [inline gateway/webhook/integration_test.go:112]: transformerURL, ok := os.Lookupenv("TEST_OVERRIDE_TRANSFORMER_URL")
	if !ok {
	    transformerURL = transformerContainer.TransformerURL
	}

Comment (lvrach) [inline gateway/webhook/integration_test.go:112]:
```suggestion
	transformerURL, ok := os.Lookupenv("TEST_OVERRIDE_TRANSFORMER_URL")
	if !ok {
	    transformerURL = transformerContainer.TransformerURL
	}
	```

File: gateway/webhook/integration_test.go [hunk 2]
```diff
@@ -175,27 +181,39 @@ func TestIntegrationWebhook(t *testing.T) {
...
+			}
 			require.NoError(t, err)
 			query.Set("writeKey", writeKey)
 
 			t.Log("Request URL:", fmt.Sprintf("%s/v1/webhook?%s", gwURL, query.Encode()))
-			req, err := http.NewRequest(http.MethodPost, fmt.Sprintf("%s/v1/webhook?%s", gwURL, query.Encode()), bytes.NewBuffer(tc.Input.Request.Body))
+			method := http.MethodPost
+			if tc.Input.Request.Method == "GET" {
+				method = http.MethodGet
+			}
```
Comment (lvrach) [inline gateway/webhook/integration_test.go:215]: simplify the logic to support all methods, defaulting to POST if empty
```suggestion
            method = tc.Input.Request.Method
			if method == "" {
			    method := http.MethodPost
			}
```

**Review by fracasula — APPROVED** [review]

**Comment by fracasula** [comment]
@sanpj2292 approved although I agree with @lvrach's comments.

**Review by lvrach — APPROVED** [review]

**Pull request merged**